File: src/main/java/com/jackchapman/adventofcode/DayOne.java
package com.jackchapman.adventofcode;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class DayOne {
    public static void main(String[] args) throws IOException {
        List<Integer> expenseReport = new ArrayList<>();
        try (InputStreamReader streamReader = new InputStreamReader(Objects.requireNonNull(DayOne.class.getClassLoader().getResourceAsStream("DayOne.txt")));
             BufferedReader reader = new BufferedReader(streamReader)) {
            String line;
            while ((line = reader.readLine()) != null) {
                expenseReport.add(Integer.parseInt(line));
            }
        }
        System.out.println("Part 1: " + find2020SumOfTwo(expenseReport));
        System.out.println("Part 2: " + find2020SumOfThree(expenseReport));
    }

    public static int find2020SumOfTwo(List<Integer> expenses) {
        for (int i : expenses) {
            for (int j : expenses) {
                if (i + j == 2020) {
                    return i * j;
                }
            }
        }
        return -1;
    }

    public static int find2020SumOfThree(List<Integer> expenses) {
        for (int i : expenses) {
            for (int j : expenses) {
                for (int k : expenses) {
                    if (i + j + k == 2020) {
                        return i * j * k;
                    }
                }
            }
        }
        return -1;
    }
}

File: src/main/kotlin/com/jackchapman/adventofcode/DayOne.kt
package com.jackchapman.adventofcode

fun main() {
    val expenseReport: List<Int> =
        {}::class.java.classLoader.getResourceAsStream("DayOne.txt")?.bufferedReader()?.readLines()
            ?.map { it.toInt() }
            .orEmpty()
    println("Part 1: ${find2020SumOfTwo(expenseReport)}")
    println("Part 2: ${find2020SumOfThree(expenseReport)}")
}

fun find2020SumOfTwo(expenses: List<Int>): Int {
    for (i in expenses) {
        for (j in expenses) {
            if (i + j == 2020) {
                return i * j
            }
        }
    }
    return -1
}

fun find2020SumOfThree(expenses: List<Int>): Int {
    for (i in expenses) {
        for (j in expenses) {
            for (k in expenses) {
                if (i + j + k == 2020) {
                    return i * j * k
                }
            }
        }
    }
    return -1
}